<?xml version="1.0"?>
<launch>
  <node pkg="tf" type="static_transform_publisher" name="tf" args="0 0 0 0 0 0 base_link usb_cam 10"/>
<!-- 
  <arg name="marker_size" default="9.0" />
	<arg name="max_new_marker_error" default="0.05" />
	<arg name="max_track_error" default="0.05" />
        
	<arg name="cam_image_topic" default="/usb_cam/image_raw" />
	<arg name="cam_info_topic" default="/usb_cam/camera_info" />
        
	<arg name="output_frame" default="/usb_cam" /> -->
	<!-- <arg name="bundle_files" default="$(find ar_track_alvar)/bundles/truthTableLeg.xml $(find ar_track_alvar)/bundles/table_8_9_10.xml" /> -->

	<!-- <node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen">
		<param name="marker_size"           type="double" value="$(arg marker_size)" />
		<param name="max_new_marker_error"  type="double" value="$(arg max_new_marker_error)" />
		<param name="max_track_error"       type="double" value="$(arg max_track_error)" />
		<param name="output_frame"          type="string" value="$(arg output_frame)" /> -->
		<!-- <param name="bundle_files"          type="string" value="$(arg bundle_files)" /> -->

		<!-- <remap from="camera_image"  to="$(arg cam_image_topic)" />
		<remap from="camera_info"   to="$(arg cam_info_topic)" />
	</node> -->

  <!-- <node name="path_generator" pkg="ar_path_nav" type="path_generator_node" output="screen"/> -->
  <!-- <node name="path_tracker" pkg="ar_path_nav" type="path_tracker_node" output="screen"/> -->

  <node pkg="rf2o_laser_odometry" type="rf2o_laser_odometry_node" name="rf2o_laser_odometry" output="screen">
    <param name="laser_scan_topic" value="/scan"/>        # topic where the lidar scans are being published
    <param name="odom_topic" value="/odom" />              # topic where tu publish the odometry estimations
    <param name="publish_tf" value="true" />                   # wheter or not to publish the tf::transform (base->odom)
    <param name="base_frame_id" value="/base_link"/>            # frame_id (tf) of the mobile robot base. A tf transform from the laser_frame to the base_frame is mandatory
    <param name="odom_frame_id" value="/odom" />                # frame_id (tf) to publish the odometry estimations    
    <param name="init_pose_from_topic" value="" /> # (Odom topic) Leave empty to start at point (0,0)
    <param name="freq" value="6.0"/>                            # Execution frequency.
    <param name="verbose" value="false" />                       # verbose
  </node>

</launch>